### ISSUE TITLE
Rule idea `selector-pseudo-class-no-empty-parentheses`

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#2047]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Maybe

Invalid
```css
input:not() { }

:matches() { }
```

Valid
```css
input:not([type="submit"]) { }

:matches(section, article, aside, nav) h1 { }
```

Include `nth-child`, `nth-last-child` and etc.

> Which rule, if any, is this issue related to?

selector-pseudo-class-no-empty-selector

> What CSS is needed to reproduce this issue?

no matter

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "selector-pseudo-class-no-empty-selector": true
  }
}
```

> Which version of stylelint are you using?

no matter

> How are you running stylelint: CLI, PostCSS plugin, Node API?

no matter

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no matter

> What did you expect to happen?

no matter

> What actually happened (e.g. what warnings or errors you are getting)?

no matter

**Comment by jeddy3** [comment]
I think we might have [taken a wrong turn](https://github.com/stylelint/stylelint/tree/master/src/rules/selector-no-empty) when we started using these `*-no-empty` rules as proxies for testing the _validity_ of things. 

Previous `*-no-empty` rules (e.g. `block-no-empty`, `no-empty-source` and `comment-no-empty`) allowed a user to specify a _perference_ i.e. an empty block, comment or source is still valid CSS. `selector-no-empty` was the first `*-no-empty` rule to break this pattern.

I think we're creeping into spec _parser_ territory here, which I think is better handled in projects like [csstree](https://github.com/csstree/validator) and [css-values](https://github.com/ben-eb/css-values) and then wrapped as stylelint plugins.  These plugins are _only applicable_ to users who write spec-based CSS i.e. those who do not use language extensions introduced in PostCSS plugins or preprocessors. Such plugins would be far more powerful that these _piecemeal_ attempts to check for validity.

So, my suggestion is we close this issue in favour of someone creating a selector validator.

I also suggest we undo our mistake with `selector-no-empty` and deprecate that rule in `8.0.0`. Along with `media-feature-no-missing-punctuation` as that is a proxy for validity too.  Arguably the 
`*-no-unknown` rules belong in a plugin pack too.

Thoughts?

**Comment by ntwb** [comment]
I think you're right @jeddy3, stylelint is _*not*_ a CSS spec parser 👍

**Comment by evilebottnawi** [comment]
@jeddy3 agree, also i agree `selector-no-empty` should be deprecated.

**Comment by davidtheclark** [comment]
👍 from me, too.
